internal/provider/jwt_issuers.go

```diff
@@ -109,195 +106,190 @@ func (r *apiOidcConfigResource) Configure(
+			"Error deleting JWT Issuer",
+			fmt.Sprintf("Could not delete JWT Issuer: %s", formatAPIErrorMessage(err)),
 		)
 		return
 	}
 
-	// Remove resource from state
 	resp.State.RemoveResource(ctx)
 }
 
-func (r *apiOidcConfigResource) ImportState(
-	ctx context.Context, req resource.ImportStateRequest, resp *resource.ImportStateResponse,
-) {
-	resource.ImportStatePassthroughID(ctx, path.Root("id"), req, resp)
-}
-
-func NewApiOidcConfigResource() resource.Resource {
-	return &apiOidcConfigResource{}
-}
-
-func loadApiOidcConfigToTerraformState(
-	apiOidcConfig *client.ApiOidcConfig, state *ApiOidcConfig,
+func loadJWTIssuerToTerraformState(
+	jwtIssuer *client.JWTIssuer, state *JWTIssuer,
 ) {
-	state.ID = types.StringValue(apiOidcConfig.Id)
-	state.Audience = types.StringValue(apiOidcConfig.Audience)
-	state.Issuer = types.StringValue(apiOidcConfig.Issuer)
-	state.Jwks = types.StringValue(apiOidcConfig.Jwks)
-	state.Claim = types.StringPointerValue(apiOidcConfig.Claim)
-	state.IdentityMap = identityMapToTerraformState(apiOidcConfig.IdentityMap)
+	state.ID = types.StringValue(jwtIssuer.Id)
+	state.Audience = types.StringValue(jwtIssuer.Audience)
+	state.IssuerURL = types.StringValue(jwtIssuer.IssuerUrl)
+	state.Jwks = types.StringPointerValue(jwtIssuer.Jwks)
+	state.Claim = types.StringPointerValue(jwtIssuer.Claim)
+	state.IdentityMap = identityMapToTerraformState(jwtIssuer.IdentityMap)
 }
 
-func identityMapFromTerraformState(identityMap *[]IdentityMapEntry) *[]client.ApiOidcIdentityMapEntry {
+func identityMapFromTerraformState(identityMap *[]IdentityMapEntry) *[]client.JWTIssuerIdentityMapEntry {
 	if identityMap == nil {
 		return nil
 	}
-	var out []client.ApiOidcIdentityMapEntry
+	var out []client.JWTIssuerIdentityMapEntry
 	for _, mapEntry := range *identityMap {
-		out = append(out, client.ApiOidcIdentityMapEntry{
-			CcIdentity:    mapEntry.CcIdentity.ValueStringPointer(),
-			IsRegex:       mapEntry.IsRegex.ValueBoolPointer(),
+		out = append(out, client.JWTIssuerIdentityMapEntry{
 			TokenIdentity: mapEntry.TokenIdentity.ValueStringPointer(),
+			CcIdentity:    mapEntry.CcIdentity.ValueStringPointer(),
+			// TODO: Need to remove once the field has been removed from the API.
+			IsRegex: types.BoolValue(false).ValueBoolPointer(),
 		})
 	}
 	return &out
 }
 
-func identityMapToTerraformState(identityMap *[]client.ApiOidcIdentityMapEntry) *[]IdentityMapEntry {
+func identityMapToTerraformState(identityMap *[]client.JWTIssuerIdentityMapEntry) *[]IdentityMapEntry {
 	if identityMap == nil {
 		return nil
 	}
 	var out []IdentityMapEntry
 	for _, mapEntry := range *identityMap {
 		out = append(out, IdentityMapEntry{
-			CcIdentity:    types.StringPointerValue(mapEntry.CcIdentity),
-			IsRegex:       types.BoolPointerValue(mapEntry.IsRegex),
 			TokenIdentity: types.StringPointerValue(mapEntry.TokenIdentity),
+			CcIdentity:    types.StringPointerValue(mapEntry.CcIdentity),
```

Comment:
are these optional? I'm surprised we're using `StringPointerValue` here

Reply:
The `identity_map` field is optional, but if an identity map is specified, then the `token_identity` and `cc_identity` fields are required. Looks like this has not been marked in the API, so the SDK still uses `*string` for these.

I could take it up as part of the PR to address the other deferred comments from https://github.com/cockroachlabs/managed-service/pull/18531.

Reply:
Okay, Ideally it wouldn't be a pointer if the value is always there since it just makes the sdk more annoying to use.  But I get it based on the state we're in now.  If you end up fixing it later, that would be nice.